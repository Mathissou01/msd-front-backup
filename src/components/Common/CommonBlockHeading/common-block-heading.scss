@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.z-index" as index;

.c-CommonBlockHeading {
  z-index: index.z("site-content");
  display: flex;
  flex-direction: column;
  gap: 10px;
  color: colors.$expert-blue;

  &__Line {
    width: 53px;
    height: 0;
    border: 2px solid colors.$brand-secondary;
    border-radius: 5px;
  }

  &_alignLeft {
    align-self: flex-start;
  }
}
