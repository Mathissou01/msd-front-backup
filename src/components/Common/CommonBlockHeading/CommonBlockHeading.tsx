import classNames from "classnames";
import "./common-block-heading.scss";

interface ICommonBlockHeadingProps {
  titleContent: string;
  isAlignLeft?: boolean;
}

export default function CommonBlockHeading({
  titleContent,
  isAlignLeft = false,
}: ICommonBlockHeadingProps) {
  const headingClassNames = classNames("c-CommonBlockHeading", {
    "c-CommonBlockHeading_alignLeft": isAlignLeft,
  });
  return (
    <div className={headingClassNames}>
      <h2>{titleContent}</h2>
      <div className="c-CommonBlockHeading__Line" />
    </div>
  );
}
