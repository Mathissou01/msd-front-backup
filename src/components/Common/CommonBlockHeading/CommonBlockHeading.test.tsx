import { render } from "@testing-library/react";
import CommonBlockHeading from "./CommonBlockHeading";

const mock = {
  title: "common block heading",
};

it("renders", () => {
  const { container } = render(<CommonBlockHeading blockTitle={mock.title} />);
  expect(container).toMatchSnapshot();
});
