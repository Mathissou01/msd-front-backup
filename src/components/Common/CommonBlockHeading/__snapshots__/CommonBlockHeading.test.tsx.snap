// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonBlockHeading renders 1`] = `
<div>
  <div
    class="c-CommonBlockHeading"
  >
    <h2>
      common block heading
    </h2>
    <div
      class="c-CommonBlockHeading__Line"
    />
  </div>
</div>
`;
